parsers/mcmaster-parser-2.js

```diff
@@ -18,6 +18,68 @@ function convertTo24Hour(time) {
     return time.replace(/(AM|PM)/, '');
 }
 
+
+/**
+ * Takes any date and turns it into the correct format, so javascript gets the right date.
+ * // MM/DD/YYYY is the format the javascript object reads or YYYY/MM/DD
+ * @param {String} startDate,{String} endDate
+ * @return {String Array} [startDate,endDate]
+ */
+
+function fixDateFormat(sem_start,sem_end){
+
+    if (sem_start.indexOf('-') > -1) {
+        startDate = sem_start;
+        endDate = sem_end;
+        return [startDate,endDate]
+    }
+    //assume format is either YYYY/MM/DD TO START OR MM/DD/YYYY
+    startDateSplit = sem_start.split("/");
+    endDateSplit = sem_end.split("/");
+
+    startMonth = parseInt(startDateSplit[0]);
+    startDay = parseInt(startDateSplit[1]);
+    startYear = parseInt(startDateSplit[2]);
+
+    endMonth = parseInt(endDateSplit[0]);
+    endDay = parseInt(endDateSplit[1]);
+    endYear = parseInt(endDateSplit[2]);
+
+    // This means the format is YYYY/MM/DD, because the 0th index is a year
+    if(startDateSplit[0].length == 4 && endDateSplit[0].length == 4){
+        startDate = sem_start;
+        endDate = sem_end;
+        return [startDate,endDate]
+
+    } else if (startMonth > 12 || endMonth > 12 ||
+        (startYear === endYear && startMonth > endMonth)){
+        // It must be DD/MM/YY, because month is never >12, so switch the day and month index
```

Comment:
This comment could simply say: "Date is DD/MM/YYYY because month can't be > 12", and then the functon's comment can say that the dates returned are of the format MM/DD/YYYY (always)